### PR TITLE
Standardize how internal slots are named and linked to.

### PR DESCRIPTION
**Pull request by taylor-b**
Related to #1112.

This replaces any instance of `[[<dfn>fooBar</dfn>]]` with
`<dfn>[[\FooBar]]</dfn>`, which means we don't have to worry about
definitions conflicting, and can have a slot whose name matches
an attribute without `<dfn>`s conflicting.

Also changed all internal slot names to be UpperCamelCase, for consistency and because this matches EcmaScript.

***
[Preview](https://s3.amazonaws.com/pr-preview/taylor-b/webrtc-pc/internal_slots.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/webrtc-pc/4e42d60...taylor-b:58909ca.html)

### CODE REVIEW

**Comment by burnburn** [comment]
@dontcallmedom can you give an opinion on casing, a la issue #1360 ?

**Comment by dontcallmedom** [comment]
since [EcmaScript](https://tc39.github.io/ecma262/) uses CapitalizedCase for internal slots, and since that's where that notion is grounded, I would suggest we adopt this.

**Comment by adam-be** [comment]
Nice that you found a solution to the attribute and slot name collision issue @taylor-b. Should we go for initial upper case before merging this?

One other thing I was thinking about is when slots have very short and generic names; like [[\Id]] for example. It might make sense to add some more information to that name to avoid future name collisions and give a hint about its origin. Like [[\DataChannelId]] for example.

File: webrtc.html [hunk 1]
```diff
@@ -8419,98 +8403,90 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
...
                 <li>
-                  <p>Let <var>channel</var> have an [[<dfn id=
-                  "maxRetransmits-slot">MaxRetransmits</dfn>]] internal slot
+                  <p>Let <var>channel</var> have a <dfn>[[\maxRetransmits]]</dfn> internal slot
                   initialized to <var>option</var>'s <code>maxRetransmits</code>
                   member, if present, otherwise <code>null</code>.</p>
                 </li>
                 <li>
-                  <p>Let <var>channel</var> have an [[<dfn>DataChannelId</dfn>]]
+                  <p>Let <var>channel</var> have an <dfn>[[\id]]</dfn>
```
Comment (adam-be) [inline webrtc.html]: Slot name too generic?

**Comment by taylor-b** [comment]
> Should we go for initial upper case before merging this?

I don't see why not.

> It might make sense to add some more information to that name to avoid future name collisions and give a hint about its origin. Like [[\DataChannelId]] for example.

Sounds good to me. Are there any others that are too ambiguous (like "protocol")?

**Comment by adam-be** [comment]
Maybe "handle" and "label".

**Comment by taylor-b** [comment]
Made some names more descriptive and switched to UpperCamelCase naming.

**Pull request merged**